pkg/config/service.go

```diff
@@ -142,7 +143,7 @@ func (s *userService) Render(
 			},
 		}
 
-		pm, d, err := r.Run(params, ctx, uc.ruleDecisions[r.ID])
+		pm, d, err := r.Run(params, ctx, uc.ruleDecisions[r.ID], generate.RandPercentage(nil))
 		if err != nil {
 			switch errors.Cause(err) {
 			case errors.ErrRuleNoMatch:
```

Comment:
We need to catch the newly introduced `ErrRuleNotInRollout` so that `Render` doesn't break here.

Reply:
And instead of straight up continuing with the loop we need to store the decision made for the rollout so we don't enroll the user in the next render.